### ISSUE TITLE
Tensors of rank 0 turn into arrays of rank 1 when .replace_with_arrays is called

### ISSUE THREAD

**Issue by Jason-S-Ross** [sympy/sympy#21870]
Minimum working example:

```python
from sympy.tensor.tensor import TensorHead
t = TensorHead("t", [])
replacement = {t(): 1}
assert t().replace_with_arrays(replacement).rank() != t.rank  # (Should be equal)
```

Testing this with tensors of rank 1 gives the expected result:

```python
from sympy import Array
from sympy.tensor.tensor import TensorHead, TensorIndexType, tensor_indices
ti = TensorIndexType("Test", dummy_name="T") 
v = TensorHead("v", [ti])
i = tensor_indices("i", ti)
replacement = {v(i): Array([0, 0])}
assert v(i).replace_with_arrays(replacement).rank() == v.rank
```

**Comment by oscarbenjamin** [comment]
CC @Upabjojr

**Comment by Jason-S-Ross** [comment]
This seems to have to do with how `Array`s are created:

```python
from sympy import Array
a0 = Array(0)
assert a0.rank() == 0
a1 = Array(a0)
assert a1.rank() == 1
a2 = Array(a1)
assert a2.rank() == 1
```

**Comment by Jason-S-Ross** [comment]
Fixed by #21875.
